resources/install.rb

```diff
@@ -57,6 +57,7 @@
     end
 
     package 'telegraf' do
+      options '-o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"'
```

Comment:
FYI, in order to make this work with `yum` you can do something similiar to what @eheydrick  did for influxdb https://github.com/bdangit/chef-influxdb/pull/130/commits/b28d22815bc40d5197d3af1946fe7b42dc4c1b31